### PR TITLE
feat: add tests

### PR DESCRIPTION
**Pull request by komarov-v**
Tests for http clients (except http-event) have been added

### CODE REVIEW

**Comment by komarov-v** [comment]
Since prettier doesn't want work well with my WebStorm I decided to add `eslint-prettier-plugin` which run code format checks as `eslint`rule. Also this plugin provides opportunity to fix style mistakes by **"Fix ESLint problems"** button.
Probably we can use `husky + lint-staged` to improve working with linters usability

File: __tests__/http-client.test.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import { FetchHttpClient } from '../src'
+
+const unmockedFetch = global.fetch
+
+beforeAll(() => {
+  global.fetch = jest.fn().mockImplementation(() =>
```
Comment (komarov-v) [inline __tests__/http-client.test.js:6]: To be honest I should have use `jest.spyOn()` but 
```javascript
jest.spyOn(global, 'fetch')
```
could't find `fetch` method in the `global object`, so I decided to mock it manually.

**Comment by komarov-v** [comment]
File             | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s                          
-----------------|---------|----------|---------|---------|--------------------------------------------
All files        |   49.11 |    48.81 |   55.42 |   49.11 |                                            
 basic-client.js |     100 |      100 |     100 |     100 |                                            
 codec.js        |     100 |      100 |     100 |     100 |                                            
 common.js       |     100 |      100 |     100 |     100 |                                            
 conference.js   |       0 |        0 |       0 |       0 | 36-338                                     
 dispatcher.js   |     100 |      100 |     100 |     100 |                                            
 error.js        |     100 |      100 |     100 |     100 |                                            
 event.js        |       0 |        0 |       0 |       0 | 9-336                                      
 http-client.js  |   84.61 |      100 |   81.81 |   84.61 | 9-10                                       
 http-event.js   |     100 |      100 |     100 |     100 |                                            
 index.js        |       0 |        0 |       0 |       0 |                                            
 mqtt-rpc.js     |   40.36 |    15.51 |   37.03 |   40.36 | 47-123,157,186-191,198-247,255-266,273-292 
 service.js      |       0 |        0 |       0 |       0 | 10-109                                     
 telemetry.js    |   94.11 |      100 |   85.71 |   94.11 | 14                                         
 tenant.js       |     100 |      100 |     100 |     100 |

**Comment by komarov-v** [comment]
Tests for http clients have been added. Test coverage of these modules has increased to 100% except of: 
*http-client.js*
```javascript
    if (!response.ok) {
      return response
        .json() // this row
        .catch(() => response.text()) // this row
        .catch(() => ({
          status: response.status,
          statusText: response.statusText,
        }))
        .then((result) => {
          throw result
        })
    }
```
*telemetry.js*
```javascript
    this.codec = new Codec(
      (data) => JSON.stringify(data),
      (_) => _ // this row
    )
````

**Review by alexkonst — APPROVED** [review]

**Pull request closed without merging**